addons/l10n_ch/models/res_bank.py

```diff
@@ -176,6 +178,18 @@ def _retrieve_l10n_ch_postal(self, iban):
             return self._pretty_postal_num(iban[-9:])
         return None
 
+    def _get_qr_code_base64(self, qr_method, amount, currency, debtor_partner, free_communication, structured_communication):
```

Comment:
Same here, I'd rather not override it, and override get_qr_vals instead so that it returns self._l10n_ch_get_qr_vals(...) when the QR method is the Swiss one.

And indeed, you might merge _l10n_ch_get_qr_vals and get_qr_vals on master

Reply:
> Same here, I'd rather not override it, and override get_qr_vals instead so that it returns self._l10n_ch_get_qr_vals(...) when the QR method is the Swiss one.

Finally, not sure if I can do that either for barcode or urls (except if I alter the functions params) since for each module the url and barcode are generated with different params;
swiss :

```
barcode = self.env['ir.actions.report'].barcode('QR', '\n'.join(qr_code_vals), width=256, height=256, quiet=1, mask="ch_cross")
```
SEPA :

```
barcode = self.env['ir.actions.report'].barcode('QR', '\n'.join(qr_code_vals), width=128, height=128, humanreadable=1)
```
same for urls... what do you think?

> And indeed, you might merge _l10n_ch_get_qr_vals and get_qr_vals on master

will do +1

Reply:
Hmmm, right. Maybe _get_qr_code_url and _get_qr_code_base64 could both call (in account) a common function returning a dictionary with the different parameters to use to generate the QR-code (_get_qr_code_generation_params ?) ? This way, we'd only override this function, and not _get_qr_code_url and _get_qr_code_base64. Also, this would have the advantage of centralizing the logic of those two functions. The result of get_qr_vals would only be one key of that dictionary.